lib/tasks/reset_permissions.rake

```diff
@@ -14,8 +14,8 @@ END_DESC
 
   task :reset => :environment do
     User.as_anonymous_admin do
-      user = User.find_or_create_by_login(:login => ENV["username"] || 'admin', :firstname => 'Admin', :lastname => 'User', :mail => Setting[:administrator])
-      src  = AuthSourceInternal.find_or_create_by_type "AuthSourceInternal"
+      user = User.where(:login => ENV["username"] || 'admin', :firstname => 'Admin', :lastname => 'User', :mail => Setting[:administrator]).first_or_create
```

Comment:
This isn't equivalent, since the old code only searched by `login` and then created the new record with all of the other attributes if it didn't exist.  Now it's searching by all of those attributes, so changing the firstname would cause it to not find the old record.